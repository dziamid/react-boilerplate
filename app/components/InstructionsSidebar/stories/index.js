import React from 'react';
import { storiesOf } from '@kadira/storybook';
import InstructionSidebar from '../index';

const listStyle = { padding: '0 0 15px', fontFamily: '"Lato", sans-serif', fontSize: '13px', fontWeight: 300, lineHeight: '17px', color: '#1F768A', letterSpacing: 0 };
const ulStyle = { listStyle: 'none', padding: 0, margin: 0, width: '230px' };
const footerInfo = {
  position: 'Senior UX/UI Designer',
  company: 'Airbnb HQ - San-Francisco',
  workflow: 'Hiring Workflow 1',

};
const body = (
  <ul style={ulStyle}>
    <li style={listStyle}>Nunc non diam metus. Fusce ornare pretium sodales.</li>
    <li style={listStyle}>Vivamus semper, tortor vel efficitur.</li>
    <li style={listStyle}>Maecenas faucibus mollis interdum.</li>
    <li style={listStyle}>Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Cras mattis consectetur pit amet fermentum.</li>
  </ul>
);

storiesOf('InstructionSidebar', module)

  .add('without footer', () => (
    <InstructionSidebar body={body} />
  ))
  .add('with footer', () => (
    <InstructionSidebar body={body} footer footerInfo={footerInfo} />
  ));
