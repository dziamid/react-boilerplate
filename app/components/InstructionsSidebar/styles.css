body {
  font-family: Lato, sans-serif;
}

.wrapper {
  float: right;
  width: 360px;
  background-color: #F5F5F5;
  height: auto;
}

.wrapper::after {
  content: '.';
  visibility: hidden;
  display: block;
  height: 0;
  clear: both;
}

.icon {
  border-radius: 50%;
  margin-bottom: 27px;
}

.icon_with_f {
  border-radius: 50%;
  margin: 0 15px 0 0;
}

.ul {
  list-style: none;
  padding: 0;
}

.title {
  width: 170px;
  text-transform: uppercase;
  font-size: 16px;
  font-weight: 300;
  color: #1F768A;
  line-height: 25px;
  letter-spacing: 0.1px;
  margin: 0 0 22px;
}

.title_with_p {
  width: 190px;
  text-transform: uppercase;
  font-size: 16px;
  font-weight: 300;
  color: #1F768A;
  line-height: 24px;
  letter-spacing: 0.1px;
  margin: 0 0 25px;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 25px 0 45px 50px;
}

.container_bot {
  display: flex;
  flex-direction: column;
}

.container_with_f {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  padding: 40px 0 35px 20px;
}
