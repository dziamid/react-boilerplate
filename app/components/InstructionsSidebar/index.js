import React from 'react';
import styles from './styles.css';

export default class InstructionsSidebar extends React.Component {

  render() {
    const {
      footer,
      footerInfo,
      body,
    } = this.props;

    const footerStyles = {
      wrapper: { backgroundColor: '#DDF6F6', fontFamily: '"Lato", sans-serif', fontSize: '13px', lineHeight: '11px', padding: '0 0 30px' },
      container: { display: 'flex', flexDirection: 'column', alignItems: 'flex-start', padding: '0 0 0 20px' },
      title: { display: 'inline-block', fontSize: '13px', textTransform: 'uppercase', color: '#1f768a', fontWeight: 400, margin: '15px 0 20px', lineHeight: '11px' },
      ul: { listStyle: 'none', padding: 0, margin: 0, width: '200px' },
      liBold: { color: '#757575', fontWeight: 400, margin: '0 0 15px' },
      li: { color: '#757575', fontWeight: 300, margin: '0 0 15px' },
    };

    const ftr = (
      <div style={footerStyles.wrapper}>
        <div style={footerStyles.container}>
          <p style={footerStyles.title}>Summary</p>
          <ul style={footerStyles.ul}>
            <li style={footerStyles.liBold}>{footerInfo ? footerInfo.position : null}</li>
            <li style={footerStyles.li}>{footerInfo ? footerInfo.company : null}</li>
            <li style={footerStyles.li}>{footerInfo ? footerInfo.workflow : null}</li>
          </ul>
        </div>
      </div>
    );

    let placeholder = 'http://placehold.it/115x115/1AC5C0/ffffff';
    let titleText = 'Tips for a winning description';
    let containerStyle = styles.container;
    let iconStyle = styles.icon;
    let titleStyle = styles.title;

    if (footer) {
      placeholder = 'http://placehold.it/40x40/1AC5C0/ffffff';
      containerStyle = styles.container_with_f;
      iconStyle = styles.icon_with_f;
      titleText = 'Optimize Job Position Requirements';
      titleStyle = styles.title_with_p;
    }

    return (
      <div className={styles.wrapper}>
        <div className={containerStyle}>
          <div>
            <img className={iconStyle} src={placeholder} alt="icon" />
          </div>
          <div className={styles.container_bot}>
            <h1 className={titleStyle}>{titleText}</h1>
            { body }
          </div>
        </div>
        { footer ? ftr : null}
      </div>
    );
  }
}
