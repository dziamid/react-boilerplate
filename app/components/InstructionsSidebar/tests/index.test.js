import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import InstructionsSidebar from '../index';
import injectTapEvent from 'utils/react-tap-event-plugin';
injectTapEvent();

const summary = {
  position: 'Senior UX/UI Designer',
  company: 'Airbnb HQ - San-Francisco',
  workflow: 'Hiring Workflow 1',

};

describe('<InstructionsSidebar />', () => {
  it('should render InstructionsSidebar without summary', () => {
    const wrapper = shallow(
      <InstructionsSidebar />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render InstructionsSidebar with summary', () => {
    const wrapper = shallow(
      <InstructionsSidebar summary={summary} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
