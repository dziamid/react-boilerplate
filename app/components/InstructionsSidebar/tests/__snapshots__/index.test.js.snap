exports[`<InstructionsSidebar /> should render InstructionsSidebar with summary 1`] = `
<div>
  <div>
    <div>
      <img
        alt="icon"
        src="http://placehold.it/40x40/1AC5C0/ffffff" />
    </div>
    <div>
      <h1>
        Optimize Job Position Requirements
      </h1>
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#DDF6F6",
        "fontFamily": "\"Lato\", sans-serif",
        "fontSize": "13px",
        "lineHeight": "11px",
        "padding": "0 0 30px",
      }
    }>
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "column",
          "padding": "0 0 0 20px",
        }
      }>
      <p
        style={
          Object {
            "color": "#1f768a",
            "display": "inline-block",
            "fontSize": "13px",
            "fontWeight": 400,
            "lineHeight": "11px",
            "margin": "15px 0 20px",
            "textTransform": "uppercase",
          }
        }>
        Summary
      </p>
      <ul
        style={
          Object {
            "listStyle": "none",
            "margin": 0,
            "padding": 0,
            "width": "200px",
          }
        }>
        <li
          style={
            Object {
              "color": "#757575",
              "fontWeight": 400,
              "margin": "0 0 15px",
            }
          }>
          Senior UX/UI Designer
        </li>
        <li
          style={
            Object {
              "color": "#757575",
              "fontWeight": 300,
              "margin": "0 0 15px",
            }
          }>
          Airbnb HQ - San-Francisco
        </li>
        <li
          style={
            Object {
              "color": "#757575",
              "fontWeight": 300,
              "margin": "0 0 15px",
            }
          }>
          Hiring Workflow 1
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<InstructionsSidebar /> should render InstructionsSidebar without summary 1`] = `
<div>
  <div>
    <div>
      <img
        alt="icon"
        src="http://placehold.it/115x115/1AC5C0/ffffff" />
    </div>
    <div>
      <h1>
        Tips for a winning description
      </h1>
    </div>
  </div>
</div>
`;
